### PR TITLE
Add guild create endpoint

### PR DESCRIPTION
**Pull request by perryprog**
10 years later.... I make the pr

### CODE REVIEW

File: lib/coxir/struct/guild.ex [hunk 1]
```diff
@@ -34,6 +34,31 @@ defmodule Coxir.Struct.Guild do
...
+  - `explicit_content_filter` - the explicit content filter level in the created guild
+  - `roles` - the roles in the created guild
+  - `channels` - the channels in the created guild
+
+  Refer to [this](https://discordapp.com/developers/docs/resources/guild#create-guild)
+  for a broader explanation on the fields and their defaults.
+  """
+  @spec create(Enum.t) :: map
+  def create(params) do
+    API.request(:POST, "/guilds", params)
```
Comment (Xh4H) [inline lib/coxir/struct/guild.ex]: What about following the function style of https://github.com/satom99/coxir/blob/master/lib/coxir/struct/guild.ex#L81 for functions containing one single line of code?

**Review by Xh4H — CHANGES_REQUESTED** [review]

File: lib/coxir/struct/guild.ex (hunk 1 above)
Comment (perryprog) [reply lib/coxir/struct/guild.ex]: I believe we only do this when we are referencing another function. There's a few times this happens below from where I linked.

https://github.com/satom99/coxir/blob/master/lib/coxir/struct/guild.ex#L178

**Comment by Xh4H** [comment]
And a question to the lib maintainer, should we implement this in another place as well taking into account that it doesn't take a Guild as parameter?

**Comment by perryprog** [comment]
This should be considered. It'd somewhat make sense in Client I suppose.

File: lib/coxir/struct/guild.ex (hunk 1 above)
Comment (Xh4H) [reply lib/coxir/struct/guild.ex]: If you check this PR you will see we followed the style we mentioned. https://github.com/satom99/coxir/pull/5/files

Comment (perryprog) [reply lib/coxir/struct/guild.ex]: Yes, that's what I mean.

When we have a function just act as a wrapper for another function,we use the `do:` syntax.  For example:
```elixir
  def edit(%{id: id}, params),
    do: edit(id, params)

  def edit(guild, params) do
    API.request(:patch, "guilds/#{guild}", params)
    |> pretty
  end
```

That PR there is somewhat similar, but it's pretty borderline and fine either way. The rest of the codebase when having a one line function that isn't a wrapper on-top of another function does not use the `do:` syntax. For example:
```elixir
  def delete(%{id: id}),
    do: delete(id)

  def delete(guild) do
    API.request(:delete, "guilds/#{guild}")
  end

  def create_channel(%{id: id}, params),
    do: create_channel(id, params)

  def create_channel(guild, params) do
    API.request(:post, "guilds/#{guild}/channels", params)
  end

  def bulk_delete_messages(%{id: id}, messages),
    do: bulk_delete_messages(id, messages)

  def bulk_delete_messages(channel, messages) do
    API.request(:post, "channels/#{channel}/messages/bulk-delete", %{messages: messages})
  end

  def typing(%{id: id}),
    do: typing(id)

  def typing(channel) do
    API.request(:post, "channels/#{channel}/typing")
  end

  # and so on 
```

**Comment by Xh4H** [comment]
It would be nice if only we had a `Client`.

**Comment by perryprog** [comment]
Oh my bad, I meant `User`.

**Comment by Xh4H** [comment]
Same issue, User functions expect a User as first parameter.

**Pull request merged**